src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -96,16 +108,44 @@
     /** EntityResolver provider, used in SLD parsing */
     EntityResolverProvider entityResolverProvider;
 
+    /** HTTP client used to fetch remote styles * */
+    CloseableHttpClient httpClient;
     /**
      * This flags allows the kvp reader to go beyond the SLD library mode specification and match
      * the first style that can be applied to a given layer. This is for backwards compatibility
      */
     private boolean laxStyleMatchAllowed = true;
 
     public GetMapKvpRequestReader(WMS wms) {
+        this(wms, null);
+    }
+
+    public GetMapKvpRequestReader(WMS wms, HttpClientConnectionManager manager) {
```

Comment:
I think this is the main problem of the current approach, the GetMapKvpRequestReader is only initialized once at startup, past that it won't be re-created and thus changes to the chaching configuration will be ignored (at least, I believe that what will happen). If the client needs to be preserved in order to allow caching, then it should probably be managed externally and rebuilt every time the config changes.